dotCMS/src/main/java/com/dotcms/ai/client/openai/OpenAIClient.java

```diff
@@ -103,58 +139,34 @@ public <T extends Serializable> void sendRequest(final AIRequest<T> request, fin
                 .ofNullable(payload.optString(AiKeys.MODEL))
                 .orElseThrow(() -> new DotAIModelNotFoundException("Model is not present in the request"));
         final Tuple2<AIModel, Model> modelTuple = appConfig.resolveModelOrThrow(modelName, jsonRequest.getType());
-        final AIModel aiModel = modelTuple._1;
+        //final AIModel aiModel = modelTuple._1;
 
         if (!modelTuple._2.isOperational()) {
-            request.getConfig().debugLogger(
+            appConfig.debugLogger(
                     getClass(),
                     () -> String.format("Resolved model [%s] is not operational, avoiding its usage", modelName));
             throw new DotAIModelNotOperationalException(String.format("Model [%s] is not operational", modelName));
         }
 
-
-        final long sleep = lastRestCall.computeIfAbsent(aiModel, m -> 0L)
-                + aiModel.minIntervalBetweenCalls()
-                - System.currentTimeMillis();
-        if (sleep > 0) {
-            Logger.info(
-                    this,
-                    "Rate limit:"
-                            + aiModel.getApiPerMinute()
-                            + "/minute, or 1 every "
-                            + aiModel.minIntervalBetweenCalls()
-                            + "ms. Sleeping:"
-                            + sleep);
-            Try.run(() -> Thread.sleep(sleep));
-        }
-
-        lastRestCall.put(aiModel, System.currentTimeMillis());
-
-        try (final CloseableHttpClient httpClient = HttpClients.createDefault()) {
-            final StringEntity jsonEntity = new StringEntity(payload.toString(), ContentType.APPLICATION_JSON);
-            final HttpUriRequest httpRequest = AIClient.resolveMethod(jsonRequest.getMethod(), jsonRequest.getUrl());
-            httpRequest.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
-            httpRequest.setHeader(HttpHeaders.AUTHORIZATION, "Bearer " + appConfig.getApiKey());
-
-            if (!payload.getAsMap().isEmpty()) {
-                Try.run(() -> ((HttpEntityEnclosingRequestBase) httpRequest).setEntity(jsonEntity));
-            }
-
-            try (final CloseableHttpResponse response = httpClient.execute(httpRequest)) {
-                onStreamCheckFotStatusCode(modelName, payload, response);
-
-                final BufferedInputStream in = new BufferedInputStream(response.getEntity().getContent());
-                final byte[] buffer = new byte[1024];
-                int len;
-                while ((len = in.read(buffer)) != -1) {
-                    output.write(buffer, 0, len);
-                    output.flush();
-                }
-            }
-        } catch (DotAIModelNotFoundException e) {
-            throw e;
+        try {
+            CircuitBreakerUrl.builder()
```

Comment:
Using our Circuit Breaker component instead of implementing and using duplicated logic. CircuitBreakerUrl is way more robust and offers more options to handle Http requests/responses and its probable exceptions.